morse_decoder.rb

```diff
@@ -0,0 +1,46 @@
+@code = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}
+
+def decode_char(char)
+  @code.each do |key, letter|
+    letter if char == key
+  end
+end
```

Comment:
- [ ] This is what is displayed when I try to run your methods. This is because all you are doing here is iterating over the array. You need to actually return the `letter`. Kindly do that:

  ```
  def decode_char(char)
    @code.each do |key, letter|
      return letter if char == key
    end
  end
  ```
  ![deco](https://user-images.githubusercontent.com/54848363/194652829-f1968068-848b-444a-92f6-518f520ae546.jpg)

Reply:
Thanks a lot! Done 😄